pros/serial/devices/vex/v5_device.py

```diff
@@ -87,10 +101,82 @@ class FTCompleteOptions(IntEnum):
     VEX_CRC16 = CRC(16, 0x1021)  # CRC-16-CCIT
     VEX_CRC32 = CRC(32, 0x04C11DB7)  # CRC-32 (the one used everywhere but has no name)
 
+    class SystemVersion(object):
+        class Product(IntEnum):
+            CONTROLLER = 0x11
+            BRAIN = 0x10
+
+        class BrainFlags(IntFlag):
+            pass
+
+        class ControllerFlags(IntFlag):
+            CONNECTED = 0x02
+
+        flag_map = {Product.BRAIN: BrainFlags, Product.CONTROLLER: ControllerFlags}
+
+        def __init__(self, data: tuple):
+            from semantic_version import Version
+            self.system_version = Version('{}.{}.{}-{}.{}'.format(*data[0:5]))
+            self.product = V5Device.SystemVersion.Product(data[5])
+            self.product_flags = self.flag_map[self.product](data[6])
+
+        def __str__(self):
+            return f'System Version: {self.system_version}\n' \
+                f'       Product: {self.product.name}\n' \
+                f' Product Flags: {self.product_flags.value:x}'
+
+    class SystemStatus(object):
+        def __init__(self, data: tuple):
+            from semantic_version import Version
+            self.system_version = Version('{}.{}.{}-{}'.format(*data[0:4]))
+            self.cpu0_version = Version('{}.{}.{}-{}'.format(*data[4:8]))
+            self.cpu1_version = Version('{}.{}.{}-{}'.format(*data[8:12]))
+            self.touch_version = data[12]
+            self.system_id = data[13]
+
+        def __getitem__(self, item):
+            return self.__dict__[item]
+
     def __init__(self, port: BasePort):
         self._status = None
         super().__init__(port)
 
+    class DownloadChannel(object):
```

Comment:
This should probably be above the `def __init__` for consistency

Reply:
I was going for "everything above __init__ is how to interpret V5 stuff" and "everything below __init__ modifies the V5"